src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,128 @@
-/**
- * Contains metadata relevant to a representation.
- */
-import { Quad } from 'rdf-js';
+import { Store } from 'n3';
+import { BlankNode, Literal, NamedNode, Quad, Term } from 'rdf-js';
+import { quad as createQuad, literal, namedNode } from '@rdfjs/data-model';
 
 /**
- * Metadata corresponding to a {@link Representation}.
+ * Stores the metadata triples and provides methods for easy access.
  */
-export interface RepresentationMetadata {
+export class RepresentationMetadata {
```

Comment:
Another thing: while on the one hand I like the fact that this object is triple-based, on the other hand I'm wondering if this is not overkill.
What is the reason for representing this in RDF in memory? Is there a real need?
Wouldn't POJO's be more efficient, with optional RDF serialization afterwards if needed?

Reply:
Because it was designed like that by @RubenVerborgh in the architecture ;) .

We need the triples anyway, and this way there is consistency of how everything is used everywhere though. Unless you mean internally storing everything in an object instead of an n3 store and only serializing when `quads` is called. That's also an option but I'm not sure how much difference that would make. Definitely want to keep using those constants for keys, as that is the easiest way to keep this generic.

Reply:
> Unless you mean internally storing everything in an object instead of an n3 store and only serializing when quads is called. 

Yes, that's what I meant.

The n3store index is the part that may be overkill, since we only really need indexing by key, which can be solved by a simple hash. No strong opinion on this though, the overhead should be very small in any case.,

Reply:
Personally I prefer using the store because otherwise several new checks would have to be added to check if a key exists or not in the internal object, while the store handles most of that behaviour automatically already.

Reply:
Could you give an example of these checks? Isn't it just `key in hash`?

Reply:
Well yes, but that's already going to drastically increase the size of some functions since many are oneliners ;) . And perhaps also deleting arrays again once they're empty. Also might get more complex should there be nested objects.

Reply:
I had the same thought as @rubensworks on RDF/JS, but no strong feeling.

Reply:
So after having read the whole class; yeah, make the interface RDF/JS 🙂 There's too much aliasing going on.
And then make the implementation simply inherit from `Store`.

Reply:
That would be the `Dataset` interface then to be sure?

Reply:
@rubensworks?